java-util/src/main/java/io/druid/java/util/common/io/NativeIO.java

```diff
@@ -0,0 +1,171 @@
+      initialized = true;
+    }
+    catch (NoClassDefFoundError e) {
+      LOG.info("JNA not found. Native methods will be disabled.");
+    }
+    catch (UnsatisfiedLinkError e) {
+      LOG.info("Unable to link C library. Native methods will be disabled.");
+    }
+    catch (NoSuchMethodError e) {
+      LOG.warn("Obsolete version of JNA present; unable to register C library");
+    }
+  }
+
+  public static native int posix_fadvise(int fd, long offset, long len, int flag) throws LastErrorException;
+
+  private NativeIO() {}
+
+  private static Field getFieldByReflection(Class cls, String fieldName)
+  {
+    Field field = null;
+
+    try {
+      field = cls.getDeclaredField(fieldName);
+      field.setAccessible(true);
+    }
+    catch (Exception e) {
+      // We don't really expect this so throw an assertion to
+      // catch this during development
+      assert false;
+      LOG.warn("Unable to read {} field from {}", fieldName, cls.getName());
+    }
+
+    return field;
+  }
+  /**
+   * Get system file descriptor (int) from FileDescriptor object.
+   * @param descriptor - FileDescriptor object to get fd from
+   * @return file descriptor, -1 or error
+   */
+  public static int getfd(FileDescriptor descriptor)
+  {
+    Field field = getFieldByReflection(descriptor.getClass(), "fd");
+    try {
+      return field.getInt(descriptor);
+    }
+    catch (Exception e) {
+      LOG.warn("Unable to read fd field from java.io.FileDescriptor");
+    }
+
+    return -1;
+  }
+
+  /**
+   * Remove pages from the file system page cache when they wont
+   * be accessed again
+   *
+   * @param fd     The file descriptor of the source file.
+   * @param offset The offset within the file.
+   * @param len    The length to be flushed.
+   */
+
+  public static void trySkipCache(int fd, long offset, long len)
+  {
+    if (!initialized || !fadvisePossible || fd < 0) {
+      return;
+    }
+    try {
+      // we ignore the return value as this is just best effort to avoid the cache
+      posix_fadvise(fd, offset, len, POSIX_FADV_DONTNEED);
+    }
+    catch (UnsupportedOperationException uoe) {
+      LOG.warn("posix_fadvise is not supported : ", uoe);
+      fadvisePossible = false;
+    }
+    catch (UnsatisfiedLinkError ule) {
+      // if JNA is unavailable just skipping Direct I/O
+      // instance of this class will act like normal RandomAccessFile
+      LOG.warn("Unsatisfied Link error: posix_fadvise failed on file descriptor {}, offset {} : ",
+          new Object[] {fd, offset, ule});
+      fadvisePossible = false;
+    }
+    catch (Exception e) {
+      // This is best effort anyway so lets just log that there was an
+      // exception and forget
+      LOG.warn("Unknown exception: posix_fadvise failed on file descriptor {}, offset {} : ",
+          new Object[] {fd, offset, e});
+    }
+  }
+
+
+  /**
+   * Copy from an input stream to a file minimizing cache impact. This happens chunk by chunk
```

Comment:
suggest modifying first line to:

`Copy from an input stream to a file minimizing cache impact on the destination.`

Reply:
Will do